lsst/ts/laser/ascii.py

```diff
@@ -54,8 +54,10 @@ def perform_magic(self, message):
         try:
             reply = self.parse_reply(self.read_until(b"\x03"))
         except TimeoutError as te:
-            reply = "Timed out"
-            raise
+            reply = None
+            self.clear()
+            self.log.error(te)
```

Comment:
I would also recommend `log.exception` here. I'm not very comfortable with skipping the exception and returning `None` though. I'd rather that be done at a higher level. But, you are probably the best judge of that.